sample-extension-plugin/src/test/java/org/opensearch/jobscheduler/sampleextension/GetScheduledJobInfoIT.java

```diff
@@ -160,38 +131,167 @@ public void testGetScheduledJobInfoByNode() throws IOException {
+    }
+
+    public void testDisableThenEnableJobInfo() throws IOException {
+        Response response = makeRequest(client(), "GET", "/_plugins/_job_scheduler/api/jobs", Collections.emptyMap(), null);
+
+        Map<String, Object> responseJson = parseResponse(response);
+
+        // verify all jobs are enabled and scheduled
+        List<Map<String, Object>> jobs = (List<Map<String, Object>>) responseJson.get("jobs");
+        Set<String> actualJobIds = new HashSet<>();
+        for (Map<String, Object> job : jobs) {
+            validateJobFields(job, true, false, actualJobIds);
+        }
+        assertEquals("All expected job IDs should be present", expectedJobIds, actualJobIds);
+        assertEquals("Should have 4 total jobs", 4, responseJson.get("total_jobs"));
+
+        disableWatcherJob("test-job-1", jobParam1);
+        disableWatcherJob("test-job-2", jobParam2);
+        disableWatcherJob("test-job-3", jobParam3);
+        disableWatcherJob("test-job-4", jobParam4);
+
+        response = makeRequest(client(), "GET", "/_plugins/_job_scheduler/api/jobs", Collections.emptyMap(), null);
+        responseJson = parseResponse(response);
+
+        assertEquals("Should have 4 total jobs", 4, responseJson.get("total_jobs"));
+
+        // Verify all jobs are disabled and descheduled
+        jobs = (List<Map<String, Object>>) responseJson.get("jobs");
+        assertNotNull("Jobs list should not be null", jobs);
+        assertEquals("Should have 4 jobs in the list", 4, jobs.size());
+
+        // Check that all expected job IDs are present and validate job fields
+        actualJobIds = new HashSet<>();
+        for (Map<String, Object> job : jobs) {
+            validateJobFields(job, false, true, actualJobIds);
+        }
+        assertEquals("All expected job IDs should be present", expectedJobIds, actualJobIds);
+
+        // verify all jobs are enabled and scheduled without duplication
+        enableWatcherJob("test-job-1", jobParam1);
+        enableWatcherJob("test-job-2", jobParam2);
+        enableWatcherJob("test-job-3", jobParam3);
+        enableWatcherJob("test-job-4", jobParam4);
+
+        response = makeRequest(client(), "GET", "/_plugins/_job_scheduler/api/jobs", Collections.emptyMap(), null);
+        responseJson = parseResponse(response);
 
-        // Validate job fields across all nodes
+        jobs = (List<Map<String, Object>>) responseJson.get("jobs");
+        assertNotNull("Jobs list should not be null", jobs);
+        assertEquals("Should have 4 jobs in the list", 4, jobs.size());
 
+        actualJobIds = new HashSet<>();
+        for (Map<String, Object> job : jobs) {
+            validateJobFields(job, true, false, actualJobIds);
+        }
+        assertEquals("All expected job IDs should be present", expectedJobIds, actualJobIds);
+    }
+
+    private void validateJobFields(
+        Map<String, Object> job,
+        boolean expectedEnabled,
+        boolean expectedDescheduled,
```

Comment:
`expectedDescheduled` is the opposite of `exceptedEnabled` in every single call. Do we need 2 separate params? When would they both be true or both be false?